src/main/java/subway/path/application/FareService.java

```diff
@@ -72,15 +63,7 @@ private int getExtraFare(List<Long> path) {
     }
 
     protected int discount(int fare, int age) {
-        if (age < CHILD_MIN_AGE) {
-            return 0;
-        }
-        if (age <= CHILD_MAX_AGE) {
-            return fare - (int) ((fare - DEDUCTION) * CHILD_DISCOUNT_RATE);
-        }
-        if (age <= TEENAGER_MAX_AGE) {
-            return fare - (int) ((fare - DEDUCTION) * TEENAGER_DISCOUNT_RATE);
-        }
-        return fare;
+        AgeGroup ageGroup = AgeGroup.getAgeGroup(age);
+        return fare - (int) ((fare - AgeGroup.getDeduction(ageGroup)) * AgeGroup.getDiscountRate(ageGroup));
```

Comment:
이왕 `enum AgeGroup`가 나왔으니까 말인데요,
```suggestion
        AgeGroup ageGroup = AgeGroup.getAgeGroup(age);
        return fare - (int) ((fare - ageGroup.getDeduction()) * ageGroup.getDiscountRate());
```
하면 더 보기 좋았을것 같아요
한 스탭 더 나가보자면
```suggestion
        AgeGroup ageGroup = AgeGroup.getAgeGroup(age);
        return ageGroup.getDetermindFare(fare);
```
같은 형태도 생각해볼법 합니다 😄

Reply:
AgeGroup에 또 ageGroup을 보내서 요청하는게 모양새가 조금 이상하다고는 생각했는데 아예 필드로 값을 넣어주면 되는 것이었네요
계산 로직도 AgeGroup으로 보내주니 좋은 것 같아요!